models/sessions.js

```diff
@@ -0,0 +1,124 @@
+var Db        = require('../config/database');
+var mongoose  = require('mongoose');
+var crypto    = require('crypto')
+
+// Generates a random string to use as a session token
+function generateToken() {
+	return crypto.randomBytes(256).toString('base64');
+}
+
+// Create the schema for a session
+var SessionSchema = mongoose.Schema({
+
+	// Defaults to a string of 256 characters
+    token	  : { type : String, default : generateToken },
+
+    // Defaults to the current time and expires in 7 days
+    createdAt : { type : Date, default : Date.now, expires : '7d' },
+
+    // Holds the ObjectId of the user logged into this session
+    userId    : { type : mongoose.Schema.Types.ObjectId, required : true}
+});
+
+// Creates a collection named sessions in MongoDB
+var SessionMongoModel = Db.model('sessions', SessionSchema);
+
+// Takes a userId as an argument and returns the token string to the callback
+function findSessionByUserId(userId, callback) {
+	SessionMongoModel.findOne({ userId : userId }, function(err, session) {
+		if (err) {
+			// TODO: Error message?
+			callback(err, null);
+			return;
+		}
+		if (session)
+			callback(null, session.token);
+		else
+			callback({
+				code	: 400,
+				message : 'The session does not exist.'
+			}, null);
```

Comment:
You don't need to specify that the second argument is null. You can just leave empty